include/wally_psbt.h

```diff
@@ -463,121 +583,467 @@ WALLY_CORE_API int wally_psbt_init_alloc(
 
 #ifndef SWIG_PYTHON
 /**
- * Free a psbt allocated by `wally_psbt_init_alloc`.
+ * Free a PSBT allocated by `wally_psbt_init_alloc`.
  *
- * :param psbt: The psbt to free.
+ * :param psbt: The PSBT to free.
  */
-WALLY_CORE_API int wally_psbt_free(struct wally_psbt *psbt);
+WALLY_CORE_API int wally_psbt_free(
+    struct wally_psbt *psbt);
 #endif /* SWIG_PYTHON */
 
 /**
- * Set the global transaction for a psbt.
- * Also initializes all of the wally_psbt_input and wally_psbt_outputs necessary
+ * Set the global transaction for a PSBT.
  *
+ * :param psbt: The PSBT to set the transaction for.
  * :param tx: The transaction to set.
- * :param psbt: The psbt to set the transaction for
+ *
+ * The global transaction can only be set on a newly created PSBT. After this
+ * call completes the PSBT will have empty inputs and outputs for each input
+ * and output in the transaction ``tx`` given.
  */
 WALLY_CORE_API int wally_psbt_set_global_tx(
     struct wally_psbt *psbt,
-    struct wally_tx *tx);
+    const struct wally_tx *tx);
 
 /**
- * Create a psbt from its serialized bytes.
+ * Create a PSBT from its serialized bytes.
  *
- * :param bytes: Bytes to create the psbt from.
- * :param bytes_len: Length of ``bytes`` in bytes.
- * :param output: Destination for the resulting psbt.
+ * :param bytes: Bytes to create the PSBT from.
+ * :param len: Length of ``bytes`` in bytes.
+ * :param output: Destination for the resulting PSBT.
  */
 WALLY_CORE_API int wally_psbt_from_bytes(
     const unsigned char *bytes,
-    size_t bytes_len,
+    size_t len,
     struct wally_psbt **output);
 
 /**
- * Get length of psbt when serialized to bytes.
+ * Get the length of a PSBT when serialized to bytes.
  *
  * :param psbt: the PSBT.
- * :param len: Length in bytes when serialized.
+ * :param written: Destination for the length in bytes when serialized.
```

Comment:
`flags` description is missing. Also for below functions

Reply:
Well spotted, thanks! I will update them in a new patch, thanks.

Reply:
Done.